### PR TITLE
Fixes nonexistant icon of tiled desert

### PR DESCRIPTION
**Pull request by UndeadNappist**
Refactoring and some cleanup as well.

Problems: Automatically reconfigures the underlying turf every time you crowbar up the tile, leading to a property where it resets the icon_state of the desert or road underneath.

In the former it sets it to something random and can even spawn grasses or, if random wildplants are added, could force them to spawn/despawn.

In the latter it sets icon_state =outermiddle.

This is caused because adding tiles literally transforms the turf into a completely different turf.

It is less of an obnoxious issue than a random purple-and-pink tile that crops every time you put floor tile down but still A Problem.

image

### CODE REVIEW

File: code/game/turfs/simulated/floor/f13_floors.dm [hunk 1]
```diff
@@ -66,6 +59,9 @@
 	//var/obj/dugpit/ground/mypit
 	//var/unburylevel = 0
 
+/turf/open/floor/plating/f13/outside/desert/New()
```
Comment (LetterN) [inline code/game/turfs/simulated/floor/f13_floors.dm]: i n i t i a l i z e

Comment (UndeadNappist) [reply code/game/turfs/simulated/floor/f13_floors.dm]: Missed a spot, thanks.

File: code/game/turfs/simulated/floor.dm [hunk 2]
```diff
@@ -97,6 +102,11 @@
 /turf/open/floor/proc/update_icon()
 	update_visuals()
 	return 1
+	if(!update_visuals())
```
[the remark this answers is not in the document]
Comment (rhicora) [reply code/game/turfs/simulated/floor.dm]: @UndeadNappist

[the remark this answers is not in the document]
Comment (AndrewL97) [reply code/game/turfs/simulated/floor.dm]: update_visuals() is a proc, not a variable, so if(!update_visuals()) checks if it the proc returned false

Its the LINDA proc to update gas overlays, etc. https://github.com/BadDeathclaw/TG-Claw/blob/d6ffeb207849abcc3d993d4326ee0cef6c9c1a73/code/modules/atmospherics/environmental/LINDA_turf_tile.dm#L97


So you:
run update_icon() super procedure (previously it would return if the super returns false)
run update_visuals, return if it returns false (why?)
do broken/burnt/icon_state checks
run update_visuals again

Are you sure you didn't mean to do if(!..()) like the original in plasteel?
image

[the remark this answers is not in the document]
Comment (UndeadNappist) [reply code/game/turfs/simulated/floor.dm]: For whatever reason I assumed you could use procs interchangeably as variables, and that if I didn't include ..() it would ignore the previous inherited traits from update_icon higher up in the hierarchy which is used in all sorts of stuff and, hopefully eventually, could be refactored all the way up to /atom/ and minimize lines of code.

[the remark this answers is not in the document]
Comment (UndeadNappist) [reply code/game/turfs/simulated/floor.dm]: Finally think I got it to an acceptable point.

File: code/game/turfs/simulated/floor.dm [hunk 3]
```diff
@@ -95,7 +100,13 @@
 	return
 
 /turf/open/floor/proc/update_icon()
+	..()
```
Comment (AndrewL97) [inline code/game/turfs/simulated/floor.dm]: Why are we now running the super proc ..()

then running it a second time on 105, and catching the return?

DM me if stuck

Comment (UndeadNappist) [reply code/game/turfs/simulated/floor.dm]: Doesn't look like it needs return 0 for any reason, so this should be fine.

File: code/game/turfs/simulated/floor/f13_floors.dm [hunk 4]
```diff
@@ -7,22 +7,16 @@
...
 	icon_state = "wasteland1"
+	icon = 'icons/turf/f13desert.dmi'
 	light_range = 3
 	light_power = 0.75
-	planetary_atmos = TRUE
 
-/turf/open/floor/plating/f13
+/turf/open/floor/plating/f13 // don't use this for anything, /f13/ is essentially just the new /unsimulated/ but for planets and should probably be phased out entirely everywhere
 	gender = PLURAL
-	name = "\proper desert"
```
Comment (LetterN) [inline code/game/turfs/simulated/floor/f13_floors.dm]: name it to instead: `name = "What the fuck mappers? why is this here"`
desc: `description = "If found, scream at the github repo about this`"

File: code/game/turfs/simulated/floor.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (LetterN) [reply code/game/turfs/simulated/floor.dm]: yes procs can be var if it returns either 0 or anything above 0
